src/netbase.h

```diff
@@ -25,6 +25,22 @@ static const int DEFAULT_CONNECT_TIMEOUT = 5000;
 //! -dns default
 static const int DEFAULT_NAME_LOOKUP = true;
 
+enum class ConnectionDirection {
+        None = 0,
+        In = (1U << 0),
+        Out = (1U << 1),
+        Both = (In | Out),
+};
+static inline ConnectionDirection& operator|=(ConnectionDirection& a, ConnectionDirection b) {
+    using underlying = typename std::underlying_type<ConnectionDirection>::type;
+    a = ConnectionDirection(underlying(a) | underlying(b));
```

Comment:
nit, `return a = ...`?

Reply:
I think it's more readable the current way